### PR TITLE
Update Python Inference Benchmark Test to Python 3.10

### PR DESCRIPTION
**Pull request by jrmccluskey**
The Python Inference benchmarks were running on a container with Python 3.8 and the upgrade to 3.9 caused a submission/runtime version mismatch on Dataflow. This updates the base container to a tensorrt image with 3.10 and updates the test as well.

The updated container has been pushed to the `apache-beam-testing` artifact registry.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Review by damccorm — APPROVED** [review]
LGTM once checks pass

File: .github/workflows/beam_Inference_Python_Benchmarks_Dataflow.yml [hunk 1]
```diff
@@ -95,7 +95,7 @@ jobs:
           arguments: |
             -PloadTest.mainClass=apache_beam.testing.benchmarks.inference.pytorch_image_classification_benchmarks \
             -Prunner=DataflowRunner \
-            -PpythonVersion=3.9 \
+            -PpythonVersion=3.10 \
```
Comment (damccorm) [inline .github/workflows/beam_Inference_Python_Benchmarks_Dataflow.yml:98]: Can we manually run these on this branch - https://github.com/apache/beam/actions/workflows/beam_Inference_Python_Benchmarks_Dataflow.yml

Comment (jrmccluskey) [reply .github/workflows/beam_Inference_Python_Benchmarks_Dataflow.yml:98]: I had to port the workflow changes to a mainline Beam repo branch, but that run is going now - https://github.com/apache/beam/actions/runs/11019905239

Comment (jrmccluskey) [reply .github/workflows/beam_Inference_Python_Benchmarks_Dataflow.yml:98]: Clean run, merging

**Pull request merged**